webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java

```diff
@@ -268,6 +276,16 @@ void updatePushRun(
     pushRunService.associatePushRunToTextUnitIds(pushRun, asset, textUnitIds);
   }
 
+  @Async
+  void scheduleAITranslation(Long repositoryId, CreateTextUnitsResult createTextUnitsResult) {
+    for (BranchStateTextUnit branchStateTextUnit : createTextUnitsResult.getCreatedTextUnits()) {
+      AITranslateJobInput aiTranslateJobInput = new AITranslateJobInput();
+      aiTranslateJobInput.setRepositoryId(repositoryId);
+      aiTranslateJobInput.setTmTextUnitId(branchStateTextUnit.getTmTextUnitId());
+      aiTranslationJobScheduler.scheduleAITranslationJob(aiTranslateJobInput);
```

Comment:
Is a separate job whose only purpose is to populate the pending MT table required here? (And one that is async?)

(The name is also a bit deceiving as I was looking to see when the job was being scheduled - but nothing was being scheduled, just pending flags set on entities in the DB)

Also - given that here we're only ever working with one repo - could we set the pending flags in the DB with just one single bulk operation, vs. doing it one by one? Doing it one go would also be fast enough to do it synchronously. (assuming a server-side insert statement vs. Hibernate still issues one insert query at a time)